br/pkg/lightning/backend/local/local.go

```diff
@@ -1761,12 +1761,28 @@ func (local *Backend) ResetEngine(ctx context.Context, engineUUID uuid.UUID) err
 		if err = local.allocateTSIfNotExists(ctx, localEngine); err != nil {
 			return errors.Trace(err)
 		}
+		failpoint.Inject("mockAllocateTSErr", func() {
+			// mock generate timestamp error when reset engine.
+			localEngine.TS = 0
+			failpoint.Return(errors.Trace(&mockGRPCErr{}))
+		})
 	}
 	localEngine.pendingFileSize.Store(0)
 
 	return err
 }
 
+type mockGRPCErr struct {
```

Comment:
`status.FromError` can be used to convert any err that's not an gRPC error to an UNKNOWN  grpc error

Reply:
cannot use mockGRPCErr (variable of type *"google.golang.org/grpc/internal/status".Status) as error value in argument to errors.Trace: *"google.golang.org/grpc/internal/status".Status does not implement error (missing method Error)compiler[InvalidIfaceAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#InvalidIfaceAssign)